### DISCUSSION TITLE
Suggestions for reducing confusiton between local and global scores in lazer.

### DISCUSSION THREAD

**Discussion by aaron-bradshaw** [ppy/osu#30846] [Ideas]
I have noticed that the separation of local and global scores in the lazer client can lead to some confusion that, in my opinion, is easily resolved. The first two bullet points outline my issues and provide some possible solutions and the third bullet point summarizes why these issues are a problem. The fourth bullet point shows a more rudimentary fix that would not involve any major UI design changes and would be completely optional for all players. This conveniently solves both issues until a more fleshed out solution can be applied to the current UI or added into a future UI redesign.

1. My first issue, marked "1" in the first screnshot, is the absence of a rank indicator showing that I have a previous score on this difficulty. This indicator already exists for other scores, as shown by the section marked "2", directly under section "1", and I clearly already have an old score on this difficulty, as shown in the section marked "3". I do not understand why a player would want this indicator to be determined by their highest local score instead of their highest overall score.

2. Old global scores that were not imported into lazer have no representation in the "local" score leaderboard so you have to switch between panels to get the full picture of your current score standing. My suggestion for this is to either always show the "Your Personal Best" panel shown in the section marked "3", regardless of which leaderboard you are viewing. Or this could be solved by showing a "fake" local score  in the score ranking. This score would look and function identically to other local scores, but it would show the data  for a player's global score. This score would not be counted in the local rankings. Instead the "1" rank, shown in the section marked "4", would be replaced by a "G" or some other symbol to show it is a global score and not local. The local scores would then be ranked normally starting at one with the global score pinned to the top clearly separated from the rest of the local scores. 

3. These issues can easily mislead a player into believing that they are either playing a difficulty for the first time or that they have set a new top score when they, in fact, have not. 

![osulazerfeaturerequest](https://github.com/user-attachments/assets/31bc68b1-47ad-46c1-a482-3524ca452c4e)

4. I believe that all of these isssues could be solved by adding a check box to automatically import global scores for a mapset whenever you download it from the website. This is shown in the screenshot below in the section marked "5". There still would be no replay, but the basic raw score data is already shown in section "3" of the screenshot above, so why not import it to provide clear and consistent communication?

![osulazerfeaturerequest2](https://github.com/user-attachments/assets/6193fa6e-828c-4f59-b9dc-162e827e0197)
P.S. I searched issues and discussions for "global" to find a similar post but I did not see one. Apologies if this is a duplicate request.

**Comment by peppy** [comment]
1 seems like a bug.
2 i tend to agree with these days. it would be great to automatically pull online scores into local scores on beatmap download/import.
